backend/controllers/auth_controller.py

```diff
@@ -32,11 +34,29 @@ def register(data):
 
         user = user_class(name=data['name'], email=email, password=data['password'])
         user.save_to_db()
+
+        if role == 'student':
+            student_data = {
+                "student_id": data.get("student_id"),  # Ensure this data is provided in the request
+                "name": data['name'],
+                "email": email,
+                "bio": data.get("bio", ""),  # Optional bio
+                "courses_enrolled": [],  # Initialize as empty, can be updated later
+                "profile_image": data.get("profile_image", "")  # Optional image path
+            }
+
+            # Create student profile using the StudentController
+            student_response = StudentController.create_student_profile(student_data)
```

Comment:
i think this is not the best practice. what it does is it creates a new database to store the info just for student profile, and this way we're gonna need another one for instructors.